python/ray/air/checkpoint.py

```diff
@@ -22,6 +22,7 @@
 from ray.util.ml_utils.filelock import TempFileLock
 
 _DICT_CHECKPOINT_FILE_NAME = "dict_checkpoint.pkl"
+_METADATA_CHECKPOINT_FILE_NAME = "metadata_checkpoint.pkl"
```

Comment:
The changes in this file is purely to make the following scenario work:

For checkpoint conversion, what is the expected behavior say if I start with a directory checkpoint (1), convert it to dict (2), add some customer marker to it by dict[“foo”] = “bar” (3), convert it back to directory checkpoint (4) and then convert it back to dict checkpoint (5). Now if I check the marker, is the marker supposed to be there?

Reply:
This seems to be some inconsistency in the data model here.

Reply:
Captured in another PR.